enhancements/machine-config/mco-network-configuration.md

```diff
@@ -0,0 +1,253 @@
+---
+title: mco-network-configuration
+authors:
+  - "@bcrochet"
+reviewers:
+  - "@cwalters"
+approvers:
+  - TBD
+creation-date: 2020-06-25
+last-updated: 2020-06-25
+status: provisional
+---
+
+# Extend MCO to support declarative host network configuration
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The ability to configure additional host networking parameters is paramount on some platforms, especially baremetal. Baremetal nodes usally have many interfaces that need to be dynamically configured to access other, critical datacenter resources. Doing so with MachineConfig would incur reboots that are expensive in time on the baremetal platform. The ability to configure host networking parameters without rebooting is desirable. It is also desirable to rollback automatically on failure, as opposed to failing on reboot.
+
+## Motivation
+
+With hybrid clouds, node-networking setup is becoming even more challenging.
+Different payloads have different networking requirements, and not everything
+can be satisfied as overlays on top of the main interface of the node (e.g.
+SR-IOV, L2, other L2).
```

Comment:
What do you mean by 'L2, other L2'?

Reply:
Do you know how the current implementation of SR-IOV for example updates networking?

Reply:
> Do you know how the current implementation of SR-IOV for example updates networking?

I do not.

Reply:
> What do you mean by 'L2, other L2'?

That section was brought over from the original propsal. I am not the original author, so I'm not really sure what the intent was. @SchSeba would you be able to answer that?

Reply:
@yboaron for L2 I mean broadcast VLAN and other network connections for other L2 I mean protocols running on top the L2 like MPLS IGMP STP and other switching protocols

Reply:
@yboaron for sriov you will get access to the same L2 network the PF has (or you can use VLANs if the PF is connected to a TRUNK interface in the switch)

I hope this answers your questions